components/case/caseController.js

```diff
@@ -77,48 +77,87 @@ const allCaseArchives = async (req, res, next) => {
   }
 };
 
+const isCompletelyArchived = (kase) => {
+  return (
+    kase.filesCopiedToOffsiteArchiveStagingDir != null &&
+    kase.commvaultBackupJobId != null &&
+    kase.filesLoadedIntoVidarrArchival != null &&
+    kase.caseFilesUnloaded != null
+  );
+};
+
+const isNotArchived = (kase) => {
+  return (
+    kase.filesCopiedToOffsiteArchiveStagingDir == null &&
+    kase.commvaultBackupJobId == null &&
+    kase.filesLoadedIntoVidarrArchival == null &&
+    kase.caseFilesUnloaded == null
+  );
+};
+
 const addCaseArchive = async (req, res, next) => {
   try {
-    const existingCase = await caseDao.getByCaseIdentifier(
+    const existingCases = await caseDao.getByCaseIdentifier(
       req.body.caseIdentifier
     );
-    if (existingCase == null) {
-      await upsert(req.body);
+    if (existingCases == null || !existingCases.length) {
+      await upsert(req.body, true);
       res.status(201).end();
     } else {
-      if (existingCase.filesCopiedToOffsiteArchiveStagingDir == null) {
-        // no harm in modifying a case that hasn't yet been archived
-        await upsert(req.body);
-        res.status(201).end();
-        return true;
-      } else if (
-        existingCase.requisitionId == req.body.requisitionId &&
-        arraysEquals(existingCase.limsIds, req.body.limsIds) &&
-        arraysEquals(
-          existingCase.workflowRunIdsForOffsiteArchive,
-          req.body.workflowRunIdsForOffsiteArchive
-        ) &&
-        arraysEquals(
-          existingCase.workflowRunIdsForVidarrArchival,
-          req.body.workflowRunIdsForVidarrArchival
-        )
-      ) {
-        // case data is same, no need to update
-        res.status(200).end();
-      } else {
-        // case data is different but files have already been copied to archiving directory, and archiving may have begun
-        throw new ConflictingDataError(
-          `Cannot modify data for case ${existingCase.caseIdentifier} that's already been sent to the archive staging directory`
-        );
+      for (let existingCase of existingCases) {
+        if (
+          existingCase.requisitionId == req.body.requisitionId &&
+          arraysEquals(existingCase.limsIds, req.body.limsIds) &&
+          arraysEquals(
+            existingCase.workflowRunIdsForOffsiteArchive,
+            req.body.workflowRunIdsForOffsiteArchive
+          ) &&
+          arraysEquals(
+            existingCase.workflowRunIdsForVidarrArchival,
+            req.body.workflowRunIdsForVidarrArchival
+          )
+        ) {
+          // case data is same, no need to update
+          res.status(200).end();
+          return true;
+        } else if (
+          existingCase.requisitionId != req.body.requisitionId ||
+          !arraysEquals(existingCase.limsIds, req.body.limsIds)
+        ) {
+          // data has changed
+          throw new ConflictingDataError(
```

Comment:
We really _shouldn't_ in a ideal scenario. Assuming everything goes as planned a case will be closed before anything of this information is sent to Nabu and a closed case shouldn't have any changes afterwards.  
This is more to address if something happens where for example, we later realize some files weren't successfully copied to the Glacier staging directory. We want to be able to update that data while keeping a record of the first archive the case underwent. 
This would be preferred since we'll be adding a second "archiving instance" and Commvault ID vs modifying an existing one which would potentially mean recalling and resending all data to keep it in a single report.

Reply:
> we later realize some files weren't successfully copied to the Glacier staging directory

Confused as to how this would happen. If the transfer fails, wouldn't the archival process never get marked as finished? Why would it happen 'later'?

Reply:
That's right, if the transfer fails it will error. My understanding is this more of a "if humans mess things up or change things around" scenario outside of the standard archiving pipeline. 
@mlaszloffy might have a better example of when this would be useful

Reply:
I don't think a case should be able to transition from "completed" back to being "active", but if this does happen, multiple archives per case will be needed.

But yes, ideally one archive record per case.